### PR TITLE
Table View Panel

### PR DESCRIPTION
**Pull request by timio23**

### CODE REVIEW

**Review by ComfortablyCoding — CHANGES_REQUESTED** [review]
Overall looks good!

### Feedback
1. I am unable to select the `ID` field for sorting for this panel. Alternative panels (e.g. List) allow it. Was there some reasoning behind this? Screenshot 2024-08-27 at 9 51 01 AM
2. I would recommend not to have sort field as required, it should default to id.
3. I am unable to select core collections for the `Collection` input
4. See other review comments.

File: packages/table-view-panel/src/panel.vue [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+				fetchData({
+					sort_field: sort.by,
+					sort_direction: (sort.desc ? 'sort_desc' : 'sort_asc'),
+					refresh: true,
+				});
+			}
+		}
+
+		// On load
+		fetchData({
```
Comment (ComfortablyCoding) [inline packages/table-view-panel/src/panel.vue:238]: We should probably trigger this within the `onMounted` hook

**Comment by timio23** [comment]
Thanks @ComfortablyCoding for your feedback. I've updated the panel and requested an another review. Regarding your comments:

1. IDs are usually GUID now which make sorting on ID unpredictable. Saying that I recognize that we should still give people the option so I've included the ID in the sort options.
2. Sort field is now optional and defaults to the primary key field
3. This was an oversight by me. I've enabled core collections for selection.
4. I've updated the panel with these suggestion.

I noticed a bug with the relational field after saving changes to row so I added a fix and some better error handling.

**Review by ComfortablyCoding — CHANGES_REQUESTED** [review]
Thanks @timio23 ! The original feedback looks to be addressed.

While re-testing I have come across the following (final) 2 issues:
1. Sorting seems to be broken for `Disable Sorting` for `ID` field specifically, was this the unpredictable behaviour you referred to?

https://github.com/user-attachments/assets/aeca5ab8-5b0b-4240-a152-c8cdb20dfc3a

2. If a user does not have `read` permissions an unknown error is shown in the display. The expectation is a blank panel with no error will be shown.
Screenshot 2024-08-29 at 10 54 04 PM

**Comment by timio23** [comment]
Thanks @ComfortablyCoding, I've addressed the 2 points regarding sorting and permissions:
1. The sorting confusion is now fixe. I set the sorting constant to empty when "disable sort" is chosen (defaulting to the primary key).
2. The drawer-item was causing the reported issue. Adding the permissions check on the drawer-item resolved this which now shows my "No Items" warning. To show a blank panel as you expected, I had to move a few things around and I felt it still needed a notification when they don't have permissions or there are no items so I changed the v-info to v-notify in these situations. See what you think.

**Comment by timio23** [comment]
@ComfortablyCoding, those recommended changes have been made.

2. I've been testing with 8 fields which causes the "No Items" text to be off screen so I added the fixed notice to always be in view. To get the best of both, I have hidden the table's notice and changed the fixed notice to look the same and positioned it in the middle. The forbidden notice is by iteself.
3. Add to package.json

**Comment by timio23** [comment]
https://github.com/user-attachments/assets/4839cc6f-b8a2-4661-b1b1-4c5c01cbb76f

**Comment by ComfortablyCoding** [comment]
> @ComfortablyCoding, those recommended changes have been made.
> 
> 2. I've been testing with 8 fields which causes the "No Items" text to be off screen so I added the fixed notice to always be in view. To get the best of both, I have hidden the table's notice and changed the fixed notice to look the same and positioned it in the middle. The forbidden notice is by iteself.
> 3. Add to package.json

Thanks @timio23, looks great. Any chance we can get the permission notice to be at the top? It might not be viewable/obvious as to why no data is loading for tall tables otherwise.

#### Current Notice Placement
Screenshot 2024-08-30 at 11 48 35 AM

**Comment by timio23** [comment]
@ComfortablyCoding, would you prefer v-info:

image

**Comment by ComfortablyCoding** [comment]
> @ComfortablyCoding, would you prefer v-info:
> 
> image

Yes

**Comment by timio23** [comment]
@ComfortablyCoding that's sorted now

**Review by ComfortablyCoding — APPROVED** [review]
LGTM 🚀 , thanks @timio23

**Pull request merged**